### PR TITLE
First draft of rusty-workflows blog

### PR DESCRIPTION
**Pull request by nicholastmosher**

### CODE REVIEW

**Review by tjtelan — APPROVED** [review]
Most of my feedback are just suggestions w/r/t formatting and skim-reading flow. The content is good enough to publish

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 1]
```diff
@@ -0,0 +1,601 @@
+---
+title: Improving our Rusty GitHub workflows
```
Comment (tjtelan) [inline content/blog/_posts/2021-04-rusty-workflows.md:2]: This title is a good start. I think it should include "Github Actions" and "Rust" for the SEO. But what would you have searched for if you were looking to find this to solve the same problem?

**Comment by nicholastmosher** [comment]
@tjtelan thanks for the thorough review! I'm working through your comments now :)

**Comment by nicholastmosher** [comment]
I have applied some of the review comments, but it is getting late so I didn't get to all of them. I would be happy to send this out as-is, but if we haven't published it by the time I'm up tomorrow I might tweak just a little bit more.

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 2]
```diff
@@ -0,0 +1,601 @@
...
+how I tell the runner which type of machine to run the job on.
+
+You may be wondering right now, "Hey, what happened to `include` and
+`exclude`? They didn't get mixed into the matrix!", and you would be
+right. `include` and `exclude` are special keys that allow you to
+manually edit the resulting configuration. In the job above, I use the
+`include` rules to add `sccache-path: /home/runner/.cache/sccache` to
+any config that includes `os: ubuntu-latest`, but when I have
+`os: macos-latest`, I set `sccache-path: /Users/runner/Library/Caches/Mozilla.sccache`
+instead.
```
[the remark this answers is not in the document]
Comment (nicholastmosher) [reply content/blog/_posts/2021-04-rusty-workflows.md]: I went ahead and broke up this paragraph into a few smaller ones with more code snippets for illustration. I think it came out better, thanks for the tip :)

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+- Setting up `sccache` to improve our building and testing speed
+  - We actually had already set up `sccache` but it was misconfigured.
+    I'll talk about how to check that everything is set up properly.
+
+The first half of this post should be generally useful for anybody who needs
+to use GitHub Actions and wants to learn more about the build matrix. The
+second half should be useful to Rust developers who want a good starting point
+for a solid CI setup.
+
+## Using the GitHub workflows build matrix
```
Comment (ajhunyady) [inline content/blog/_posts/2021-04-rusty-workflows.md:36]: Do you need an intro on `Why Github Actions?` rather than other services?

Comment (ajhunyady) [inline content/blog/_posts/2021-04-rusty-workflows.md:36]: A title such as `From Makefiles to Github Action Workflows` would catch my attention. (not sure if this is the point of the section... as I haven't read it yet).

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+        - os: macos-latest
+          sccache-path: /Users/runner/Library/Caches/Mozilla.sccache
+      exclude:
+        - os: macos-latest
+          rust: stable
+          make:
+            name: Clippy
+```
+
+The part of this that we're interested in is the `matrix` object. Notice that
```
Comment (ajhunyady) [inline content/blog/_posts/2021-04-rusty-workflows.md:86]: Notice -> Note that

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+keys are arbitrary, you can choose any names that you want for them. I like to
+think of them as the "dimensions" of the matrix. The value at each of these keys
+must be a list:
+
+- For the key `os`, the list is `[ubuntu-latest, macos-latest]`.
+- For the key `rust`, we have a list with a single element: `[stable]`. 
+- And even though the value under `make` looks different, notice that it is still a list,
+  it is just a list of more yaml objects.
+  
+When GitHub Actions reads your job definition, it performs a sort of cross-product
```
Comment (ajhunyady) [inline content/blog/_posts/2021-04-rusty-workflows.md:97]: `a sort of` is distracting
cross-product ... what? checks?

File: content/blog/_posts/2021-04-rusty-workflows.md [hunk 6]
```diff
@@ -0,0 +1,725 @@
...
+```
+
+Ok, awesome. So now we have a strategy for adding new configurations
+as well as for tweaking options on specific configurations. If you're trying
+to consolidate a bunch of duplicate jobs, a good strategy is to start identifying
+the small pieces of each job that are different from the others, and put those
+as options in the matrix. Next I'll walk through the rest of the job definition
+and talk about how we set it up to meet our Rust project needs.
+
+## Optimizing Rust's build speed with `sccache`
```
Comment (ajhunyady) [inline content/blog/_posts/2021-04-rusty-workflows.md:294]: Change this to 2x or an exact number.

**Review by ajhunyady — APPROVED** [review]

**Comment by nicholastmosher** [comment]
I think I'm happy with it! Thanks for all the feedback, I'm going to go ahead and merge it so it's ready to publish

**Pull request merged**